src/provider/provider_os_memory.c

```diff
@@ -1160,13 +1160,9 @@ static umf_result_t os_get_ipc_handle_size(void *provider, size_t *size) {
         return UMF_RESULT_ERROR_INVALID_ARGUMENT;
     }
 
-    if (os_provider->shm_name[0]) {
-        // os_ipc_data_t->shm_name will be used
-        *size = sizeof(os_ipc_data_t);
-    } else {
-        // os_ipc_data_t->shm_name will NOT be used
-        *size = sizeof(os_ipc_data_t) - NAME_MAX;
-    }
+    // for simplicity, always return size of the full structure even if we
```

Comment:
IPC handle is send/copied between processes by the client, so with this change the `NAME_MAX` bytes will be copied every time even they are not used.

Reply:
As I wrote - this is for simplicity. Do you think we should focus on optimizing this now?

Reply:
Yes, the current test allocates the buffer for IPC handle based on umfMemoryProviderGetIPCHandleSize(provider, ..) where the "provider" is the consumer provider, not the one used in the producer.

Reply:
I will change the test so that the producer sends the IPC handle size first, so that the consumer can allocate a buffer of the correct size instead of calling umfMemoryProviderGetIPCHandleSize()

Reply:
> 

@vinser52 It makes possible receiving IPC handle of greater size (with SHM name) when the current process does not use SHM (and the IPCGetHandleSize() had returned smaller size (without SHM name) previously).

good point.